paper-server/patches/sources/net/minecraft/world/level/ServerExplosion.java.patch

```diff
@@ -234,6 +207,25 @@
          this.level.gameEvent(this.source, GameEvent.EXPLODE, this.center);
          List<BlockPos> list = this.calculateExplodedPositions();
          this.hurtEntities();
+@@ -254,6 +_,18 @@
+             this.interactWithBlocks(list);
+             profilerFiller.pop();
+         }
++        // Paper start - handle not interacts in explosion events
++        else {
++            io.papermc.paper.event.world.ExplodeEvent explodeEvent = CraftEventFactory.callExplodeEvent(this, list);
++            this.wasCanceled = explodeEvent.isCancelled();
++            this.yield = explodeEvent.getYield();
++            list.clear();
++            list.addAll(explodeEvent.blockList().stream().map(block -> ((org.bukkit.craftbukkit.block.CraftBlock) block).getPosition()).toList());
++        }
++        if (this.wasCanceled) {
++            return;
```

Comment:
You can't just return there, you need to clear the moonrise cache too, imo you can just move the check on the fire check below for now, otherwise would need to patch moonrise and add a hook which is annoying since it would only be relevant for paper

Reply:
I forget about the moonrise thing (this cuase a mess if try to remove the CB import in the start for migrate to fq..)
Then just move this to fire && !wasCancel?

Reply:
Yes, the alternative is not worth imo and if mojang add more things patch should conflict anyway